src/app/query/bitcoin/address/transactions-by-address.hooks.ts

```diff
@@ -7,28 +7,43 @@ import { sumNumbers } from '@app/common/math/helpers';
 import { useGetBitcoinTransactionsByAddressQuery } from './transactions-by-address.query';
 
 export function useBitcoinPendingTransactions(address: string) {
-  const { data: bitcoinTransactions } = useGetBitcoinTransactionsByAddressQuery(address);
-  // TODO: use useQuery select method
-  return useMemo(
-    () => (bitcoinTransactions ?? []).filter(tx => !tx.status.confirmed),
-    [bitcoinTransactions]
-  );
+  return useGetBitcoinTransactionsByAddressQuery(address, {
+    select(txs) {
+      return txs.filter(tx => !tx.status.confirmed);
+    },
+  });
+}
+
+export function useBitcoinPendingTransactionsInputs(address: string) {
+  const { data: pendingTransactions = [] } = useBitcoinPendingTransactions(address);
+
+  return useMemo(() => {
+    return pendingTransactions.flatMap(tx => tx.vin.map(input => input));
+  }, [pendingTransactions]);
+}
+
+function useBitcoinPendingTransactionsOutputs(address: string) {
+  const { data: pendingTransactions = [] } = useBitcoinPendingTransactions(address);
+
+  return useMemo(() => {
+    return pendingTransactions.flatMap(tx => {
+      const inputsFromAddress = tx.vin.filter(
+        input => input.prevout.scriptpubkey_address === address
+      );
+      // Output is possibly change, so we only subtract the value if the address
+      // is funding the tx and sending utxos to a different address
+      return tx.vout.filter(
+        output => inputsFromAddress.length && output.scriptpubkey_address !== address
+      );
+    });
+  }, [address, pendingTransactions]);
```

Comment:
Wonder if you could use a similar approach as below to filtering here, composing `select`s?

Reply:
I suppose so if the goal of every hook using a query is to still expose what the query returns, which makes sense. I can def do it that way here.